app/src/main/res/layout/key_gear_item_detail_name_section.xml

```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:orientation="vertical">
+
+    <TextView style="@style/Headline.Medium"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/base_margin_triple"
+        android:layout_marginTop="@dimen/base_margin_quintuple"
+        android:text="@string/KEY_GEAR_ITEM_VIEW_ITEM_NAME_TABLE_TITLE"
+        xmlns:android="http://schemas.android.com/apk/res/android" />
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/base_margin_triple"
+        android:layout_marginTop="@dimen/base_margin"
+        android:layout_marginEnd="@dimen/base_margin_triple">
+
+        <TextView
+            android:id="@+id/name"
+            style="@style/Body.Regular"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="Mobiltelefon"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <com.hedvig.app.ui.view.HedvigEditText
+            android:id="@+id/nameEditText"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:alpha="0"
+            android:imeOptions="actionDone"
+            android:inputType="text"
+            android:visibility="gone"
+            app:hintText="Namn"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+
+        <Button
+            android:id="@+id/addName"
+            style="@style/Body.Regular.Link"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginEnd="@dimen/base_margin_triple"
+            android:background="?attr/selectableItemBackground"
+            android:text="@string/KEY_GEAR_ITEM_VIEW_ITEM_NAME_EDIT_BUTTON"
+            android:textAllCaps="false"
+            android:textColor="@color/link_purple"
+            android:theme="@style/Body.Regular.Link"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent" />
+
+        <!--TODO translation-->
+        <Button
```

Comment:
Is this button exactly the same button as the one before it? If that is the case, we can probably re-use it instead of having two buttons with different text 🙂

Reply:
it is in the same place, but the have different onClickListeners, one shows the editText and opens the keyboard and the other one uploads the new name

Reply:
We can check which state we are in and trigger that branch in the click listener 🙂